src/wampy.js

```diff
@@ -379,12 +379,11 @@ class Wampy {
             }
         };
 
-        if (this._isPlainObject(options)) {
-            this._options = Object.assign({}, this._options, options);
-        } else if (this._isPlainObject(url)) {
-            this._options = Object.assign({}, this._options, url);
-        }
-
+        this._options = {
+            ...this._options,
+            ...(this._isPlainObject(options) ? options : {}),
+            ...(this._isPlainObject(url) ? url : {}),
+        };
```

Comment:
Well... I personally find the original version cleaner. In this case, we have 2 independent checks while they are mutually exclusive. It's not a problem here. But we even do not save LoCs...

What do you think?

Reply:
What if we kept the if/else block but changed only the `Object.assign()` part to the `...` syntax?

Reply:
Agree! Compromise! Let's make it :)